registry/registry.go

```diff
@@ -29,3 +29,14 @@ type Registry interface {
 	Records(zone string) ([]*endpoint.Endpoint, error)
 	ApplyChanges(zone string, changes *plan.Changes) error
 }
+
+//TODO(ideahitme): considering moving this to Plan
```

Comment:
the reason for not defining methods on endpoints is mostly lack of need at the moment. `FilterByOwner` would be a method to be invoked only if ownership is enabled, otherwise it is irrelevant, so IMO it should exist only in the component which requires ownership filtering. 
As for the `SameEndpoints` vs `.Equal` - totally makes sense, however endpoints comparison currently is only needed in tests, hence it is only defined in `internal/testutils`.